Source/JavaScriptCore/wasm/WasmSlowPaths.cpp

```diff
@@ -322,6 +322,37 @@ WASM_SLOW_PATH_DECL(ref_func)
     WASM_RETURN(Wasm::operationWasmRefFunc(instance, instruction.m_functionIndex));
 }
 
+WASM_SLOW_PATH_DECL(struct_new)
+{
+    auto instruction = pc->as<WasmStructNew>();
+    const auto& moduleInformation = instance->module().moduleInformation();
+    ASSERT(instruction.m_typeIndex < moduleInformation.typeCount());
+    const auto& structType = *moduleInformation.typeSignatures[instruction.m_typeIndex]->as<Wasm::StructType>();
+
+    FixedVector<uint64_t> values(structType.fieldCount());
+    for (unsigned i = 0; i < structType.fieldCount(); ++i)
+        values[i] = READ((instruction.m_firstValue - i)).encodedJSValue();
```

Comment:
We shouldn't need this intermediate FixedVector. I don't think it's buying us anything

Reply:
Cool, so I can pass a pointer to the first value, right?
Sorry for the newbie question, but how can I do this? :)

Reply:
Maybe assert that instruction.m_firstValue is on the stack, and then you should be able to pass a pointer to the first value unless I am missing something